### ISSUE TITLE
storj-sim network run error, cause node is missing ?

### ISSUE THREAD

**Issue by JunSungKang** [storj/storj#4088]
I storj local network running,  source analyzing.

i use storj shutdown `ctrl + c`, in order to computer reboot.

reboot after, storj-sim network run.
oops...! I see error 

Below is the log information.

```console
D:\jskang\003-workspace\004-github\storj\storj>storj-sim network run
redis/0                                   14:59:38.166 |                 _._
                                                       |            _.-``__ ''-._
                                                       |       _.-``    `.  `_.  ''-._           Redis 3.0.503 (00000000/0) 64 bit
                                                       |   .-`` .-```.  ```\/    _.,_ ''-._
                                                       |  (    '      ,       .-`  | `,    )     Running in standalone mode
                                                       |  |`-._`-...-` __...-.``-._|'` _.-'|     Port: 10004
                                                       |  |    `-._   `._    /     _.-'    |     PID: 17672
                                                       |   `-._    `-._  `-./  _.-'    _.-'
                                                       |  |`-._`-._    `-.__.-'    _.-'_.-'|
                                                       |  |    `-._`-._        _.-'_.-'    |           http://redis.io
                                                       |   `-._    `-._`-.__.-'_.-'    _.-'
                                                       |  |`-._`-._    `-.__.-'    _.-'_.-'|
                                                       |  |    `-._`-._        _.-'_.-'    |
                                                       |   `-._    `-._`-.__.-'_.-'    _.-'
                                                       |       `-._    `-.__.-'    _.-'
                                                       |           `-._        _.-'
                                                       |               `-.__.-'
                                                       |
redis/0                                   14:59:38.343 | [17672] 17 Apr 14:59:38.166 # Server started, Redis version 3.0.503
                                                       | [17672] 17 Apr 14:59:38.166 * The server is now ready to accept connections on port 10004
versioncontrol/0                          14:59:41.501 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\versioncontrol\\0\\config.yaml"}
versioncontrol/0                          14:59:41.509 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
                                                       | DEBUG  process/debug.go:37     debug server listening on 127.0.0.1:12009
versioncontrol/0                          14:59:41.513 | DEBUG  versioncontrol/peer.go:175      Setting version info.   {"Value": "{\"Satellite\":{\"major\":0,\"minor\":0,\"patch\":1},\"Storagenode\":{\"major\":0,\"minor\":0,\"patch
\":1},\"Uplink\":{\"major\":0,\"minor\":0,\"patch\":1},\"Gateway\":{\"major\":0,\"minor\":0,\"patch\":1},\"Identity\":{\"major\":0,\"minor\":0,\"patch\":1},\"processes\":{\"satellite\":{\"minimum\":{\"version\":\"v0.0.1\",\"url\":\"
\"},\"suggested\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"rollout\":{\"seed\":\"0000000000000000000000000000000000000000000000000000000000000001\",\"cursor\":\"\"}},\"storagenode\":{\"minimum\":{\"version\":\"v0.0.1\",\"url\":\"\"},
\"suggested\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"rollout\":{\"seed\":\"0000000000000000000000000000000000000000000000000000000000000001\",\"cursor\":\"\"}},\"storagenode-updater\":{\"minimum\":{\"version\":\"v0.0.1\",\"url\":\"
\"},\"suggested\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"rollout\":{\"seed\":\"0000000000000000000000000000000000000000000000000000000000000001\",\"cursor\":\"\"}},\"uplink\":{\"minimum\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"sug
gested\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"rollout\":{\"seed\":\"0000000000000000000000000000000000000000000000000000000000000001\",\"cursor\":\"\"}},\"gateway\":{\"minimum\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"suggested\"
:{\"version\":\"v0.0.1\",\"url\":\"\"},\"rollout\":{\"seed\":\"0000000000000000000000000000000000000000000000000000000000000001\",\"cursor\":\"\"}},\"identity\":{\"minimum\":{\"version\":\"v0.0.1\",\"url\":\"\"},\"suggested\":{\"ver
sion\":\"v0.0.1\",\"url\":\"\"},\"rollout\":{\"seed\":\"0000000000000000000000000000000000000000000000000000000000000001\",\"cursor\":\"\"}}}}"}
versioncontrol/0                          14:59:41.515 | INFO   versioncontrol/peer.go:281      Versioning server started.      {"Address": "127.0.0.1:12000"}
satellite-migration/0                     14:59:42.095 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\satellite\\0\\config.yaml"}
satellite-migration/0                     14:59:42.102 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
satellite-migration/0                     14:59:42.375 | DEBUG  migration       satellitedb/database.go:129     Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--sear
ch_path%3D%22satellite%2F0%22&application_name=satellite-migration"}
satellite-migration/0                     14:59:42.487 | INFO   migration.migrate       migrate/versions.go:218 Database Version        {"version": 147}
satellite-migration/0                     14:59:42.602 | DEBUG  migration       metainfo/config.go:167  Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%2Fmeta%22"}
satellite-garbage-collection/0            14:59:46.806 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\satellite\\0\\config.yaml"}
satellite-garbage-collection/0            14:59:46.808 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
satellite-repairer/0                      14:59:46.834 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\satellite\\0\\config.yaml"}
satellite-repairer/0                      14:59:46.836 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
satellite-core/0                          14:59:46.897 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\satellite\\0\\config.yaml"}
satellite-core/0                          14:59:46.899 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
satellite-admin/0                         14:59:46.901 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\satellite\\0\\config.yaml"}
satellite-admin/0                         14:59:46.903 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
satellite-garbage-collection/0            14:59:47.002 | DEBUG  db      satellitedb/database.go:129     Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%22&application_name=satellite-gc"}
satellite-repairer/0                      14:59:47.038 | DEBUG  db      satellitedb/database.go:129     Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%22&application_name=satellite-repairer"}
satellite/0                               14:59:47.070 | INFO   process/exec_conf.go:284        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\satellite\\0\\config.yaml"}
satellite/0                               14:59:47.072 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
satellite-admin/0                         14:59:47.097 | DEBUG  db      satellitedb/database.go:129     Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%22&application_name=satellite-admin"}
satellite-core/0                          14:59:47.102 | DEBUG  db      satellitedb/database.go:129     Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%22&application_name=satellite-core"}
satellite-admin/0                         14:59:47.108 | DEBUG  process/metrics.go:66   Telemetry disabled
satellite-admin/0                         14:59:47.123 | DEBUG  db      migrate/versions.go:144 Database version is up to date  {"version": 147}
satellite-admin/0                         14:59:47.135 | DEBUG  servers lifecycle/group.go:93   started {"items": ["debug", "admin"]}
                                                       | DEBUG  services        lifecycle/group.go:93   started {"items": ["version"]}
satellite-garbage-collection/0            14:59:47.172 | DEBUG  pointerdb       metainfo/config.go:167  Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%2Fmeta%22"}
satellite-garbage-collection/0            14:59:47.173 | INFO   satellite/gc.go:106     Version info    {"Version": "0.0.0", "Commit Hash": "", "Build Timestamp": "0001-01-01 00:00:00 +0000 UTC", "Release Build": false}
satellite-garbage-collection/0            14:59:47.177 | DEBUG  process/metrics.go:66   Telemetry disabled
satellite-garbage-collection/0            14:59:47.197 | DEBUG  db      migrate/versions.go:144 Database version is up to date  {"version": 147}
satellite-garbage-collection/0            14:59:47.198 | DEBUG  servers lifecycle/group.go:93   started {"items": ["debug"]}
satellite-garbage-collection/0            14:59:47.200 | DEBUG  services        lifecycle/group.go:93   started {"items": ["version", "metainfo:loop", "garbage-collection", "metrics"]}
satellite-repairer/0                      14:59:47.212 | DEBUG  pointerdb       metainfo/config.go:167  Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%2Fmeta%22"}
satellite-repairer/0                      14:59:47.216 | INFO   satellite/repairer.go:103       Version info    {"Version": "0.0.0", "Commit Hash": "", "Build Timestamp": "0001-01-01 00:00:00 +0000 UTC", "Release Build": false}
satellite-repairer/0                      14:59:47.217 | DEBUG  process/metrics.go:66   Telemetry disabled
satellite-repairer/0                      14:59:47.231 | DEBUG  db      migrate/versions.go:144 Database version is up to date  {"version": 147}
satellite-repairer/0                      14:59:47.232 | DEBUG  servers lifecycle/group.go:93   started {"items": ["debug"]}
satellite-repairer/0                      14:59:47.233 | DEBUG  services        lifecycle/group.go:93   started {"items": ["version", "overlay", "orders:chore", "repair"]}
satellite/0                               14:59:47.257 | DEBUG  db      satellitedb/database.go:129     Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%22&application_name=satellite-api"}
satellite-core/0                          14:59:47.273 | DEBUG  pointerdb       metainfo/config.go:167  Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%2Fmeta%22"}
satellite-core/0                          14:59:47.285 | INFO   satellite/core.go:176   Version info    {"Version": "0.0.0", "Commit Hash": "", "Build Timestamp": "0001-01-01 00:00:00 +0000 UTC", "Release Build": false}
satellite-core/0                          14:59:47.289 | DEBUG  process/metrics.go:66   Telemetry disabled
satellite-core/0                          14:59:47.306 | DEBUG  db      migrate/versions.go:144 Database version is up to date  {"version": 147}
satellite-core/0                          14:59:47.308 | DEBUG  servers lifecycle/group.go:93   started {"items": ["debug"]}
satellite-core/0                          14:59:47.318 | DEBUG  services        lifecycle/group.go:93   started {"items": ["version", "contact:service", "overlay", "overlay:dq-stray-nodes", "overlay", "metainfo:loop", "repair:checke
r", "audit:worker", "audit:chore", "expireddeletion:chore", "accounting:tally", "accounting:rollup", "accounting:project-bw-rollup", "accounting:rollup-archive", "payments.stripe:service", "gracefulexit", "metrics"]}
                                                       | INFO   payments.stripe:clearing        stripecoinpayments/clearing.go:61       running account balance update cycle
                                                       | INFO   payments.stripe:clearing        stripecoinpayments/clearing.go:50       running transactions update cycle
satellite-core/0                          14:59:47.373 | INFO   accounting:rollup       rollup/rollup.go:92     RollupStats is empty
satellite-core/0                          14:59:47.401 | DEBUG  payments.stripe:service stripecoinpayments/service.go:182       no transactions found, skipping update
satellite/0                               14:59:47.471 | DEBUG  pointerdb       metainfo/config.go:167  Connected to:   {"db source": "postgres://postgres:Wnstjddk1gkwk!@127.0.0.1:5432/postgres?sslmode=disable&options=--search_path%
3D%22satellite%2F0%2Fmeta%22"}
satellite/0                               14:59:47.484 | INFO   satellite/api.go:198    Version info    {"Version": "0.0.0", "Commit Hash": "", "Build Timestamp": "0001-01-01 00:00:00 +0000 UTC", "Release Build": false}
satellite/0                               14:59:47.581 | DEBUG  console:endpoint        consoleweb/server.go:137        Starting Satellite UI.  {"Address": "127.0.0.1:10002"}
satellite/0                               14:59:47.587 | DEBUG  process/metrics.go:66   Telemetry disabled
satellite/0                               14:59:47.601 | DEBUG  db      migrate/versions.go:144 Database version is up to date  {"version": 147}
satellite/0                    12pm47QHZM 14:59:47.602 | INFO   satellite/api.go:238    Node 12pm47QHZMxJrZibR9M7eXTx4dBzcJYg5d9cP3WhX1Qirwo8Jbk started
satellite/0                    12pm47QHZM 14:59:47.602 | DEBUG  servers lifecycle/group.go:93   started {"items": ["debug", "server", "console:endpoint"]}
                                                       | INFO   satellite/api.go:239    Public server started on 127.0.0.1:10000
                                                       | INFO   satellite/api.go:240    Private server started on 127.0.0.1:10001
satellite/0                    12pm47QHZM 14:59:47.605 | DEBUG  services        lifecycle/group.go:93   started {"items": ["version", "overlay", "contact:service", "orders:chore", "metainfo:piecedeletion", "metainfo:endpoint", "mail
:service", "payments.stripe:version"]}
                                                       | DEBUG  payments.stripe:version stripecoinpayments/version.go:43        running conversion rates update cycle
                                                       | INFO   payments.stripe:service stripecoinpayments/service.go:380       Coinpayment client is missing public key
gateway/0                                 14:59:49.949 | INFO   process/exec_conf.go:277        Configuration loaded    {"Location": "C:\\Users\\kjuns\\AppData\\Roaming\\Storj\\Local-Network\\gateway\\0\\config.yaml"}
gateway/0                                 14:59:49.952 | DEBUG  process/tracing.go:74   Anonymized tracing disabled
gateway/0                                 14:59:49.953 | DEBUG  process/metrics.go:66   Telemetry disabled
gateway/0                                 14:59:49.954 | INFO   gateway@v1.1.9/main.go:150      Starting Tardigrade S3 Gateway
                                                       |
                                                       |
                                                       | INFO   gateway@v1.1.9/main.go:151      Endpoint: 127.0.0.1:11000
                                                       |
                                                       | INFO   gateway@v1.1.9/main.go:152      Access key: 4FVxBVqqTjYMu2Av8kAhycQHENAG
                                                       |
                                                       | INFO   gateway@v1.1.9/main.go:153      Secret key: 3pGgXafhpMrWidxW4P6iCkzTgtCV
                                                       |
                                                       | DEBUG  process/debug.go:37     debug server listening on 127.0.0.1:11009
gateway/0                                 14:59:50.099 |
                                                       | API: SYSTEM()
                                                       | Time: 14:59:50 KST 04/17/2021
                                                       | DeploymentID: e592aad6-8b85-4da6-adca-8eed6ec8dfef
                                                       | Error: WARNING: Expected number of all hosts (0) to be remote +1 (0) (*errors.errorString)
                                                       |        4: C:\Users\kjuns\go\pkg\mod\github.com\minio\minio@v0.0.0-20201028162317-be7f67268d1e\cmd\peer-rest-client.go:866:cmd.newPeerRestClients()
                                                       |        3: C:\Users\kjuns\go\pkg\mod\github.com\minio\minio@v0.0.0-20201028162317-be7f67268d1e\cmd\notification.go:1292:cmd.NewNotificationSys()
                                                       |        2: C:\Users\kjuns\go\pkg\mod\github.com\minio\minio@v0.0.0-20201028162317-be7f67268d1e\cmd\server-main.go:160:cmd.newAllSubsystems()
                                                       |        1: C:\Users\kjuns\go\pkg\mod\github.com\minio\minio@v0.0.0-20201028162317-be7f67268d1e\cmd\gateway-main.go:304:cmd.StartGateway()
satellite-core/0                          14:59:52.799 | ERROR  repair:checker  checker/checker.go:140  IdentifyInjuredSegments error   {"error": "checker error: could not get estimate of total number of nodes: checker error: segmen
t health is meaningless: there are no nodes", "errorVerbose": "checker error: could not get estimate of total number of nodes: checker error: segment health is meaningless: there are no nodes\n\tstorj.io/storj/satellite/repair/check
er.(*checkerObserver).RemoteSegment:329\n\tstorj.io/storj/satellite/metainfo.(*observerContext).RemoteSegment:128\n\tstorj.io/storj/satellite/metainfo.handlePointer:324\n\tstorj.io/storj/satellite/metainfo.iterateDatabase.func2:406\
n\tstorj.io/storj/storage/postgreskv.(*Client).IterateWithoutLookupLimit:233\n\tstorj.io/storj/satellite/metainfo.iterateDatabase:374\n\tstorj.io/storj/satellite/metainfo.(*Loop).runOnce:261\n\tstorj.io/storj/satellite/metainfo.(*Lo
op).Run:220\n\tstorj.io/storj/private/lifecycle.(*Group).Run.func2.1:81\n\truntime/pprof.Do:40\n\tstorj.io/storj/private/lifecycle.(*Group).Run.func2:80\n\tgolang.org/x/sync/errgroup.(*Group).Go.func1:57"}
                                                       | storj.io/storj/satellite/repair/checker.(*Checker).IdentifyInjuredSegments
                                                       |        D:/jskang/003-workspace/004-github/storj/storj/satellite/repair/checker/checker.go:140
                                                       | storj.io/common/sync2.(*Cycle).Run
                                                       |        C:/Users/kjuns/go/pkg/mod/storj.io/common@v0.0.0-20210217105242-970e119468ed/sync2/cycle.go:92
                                                       | storj.io/storj/satellite/repair/checker.(*Checker).Run.func1
                                                       |        D:/jskang/003-workspace/004-github/storj/storj/satellite/repair/checker/checker.go:77
                                                       | golang.org/x/sync/errgroup.(*Group).Go.func1
                                                       |        C:/Users/kjuns/go/pkg/mod/golang.org/x/sync@v0.0.0-20201207232520-09787c993a3a/errgroup/errgroup.go:57
satellite-core/0                          14:59:52.957 | ERROR  services        lifecycle/group.go:87   unexpected shutdown of a runner {"name": "repair:checker", "error": "checker error: could not get estimate of total number of no
des: checker error: segment health is meaningless: there are no nodes", "errorVerbose": "checker error: could not get estimate of total number of nodes: checker error: segment health is meaningless: there are no nodes\n\tstorj.io/st
orj/satellite/repair/checker.(*checkerObserver).RemoteSegment:329\n\tstorj.io/storj/satellite/metainfo.(*observerContext).RemoteSegment:128\n\tstorj.io/storj/satellite/metainfo.handlePointer:324\n\tstorj.io/storj/satellite/metainfo.
iterateDatabase.func2:406\n\tstorj.io/storj/storage/postgreskv.(*Client).IterateWithoutLookupLimit:233\n\tstorj.io/storj/satellite/metainfo.iterateDatabase:374\n\tstorj.io/storj/satellite/metainfo.(*Loop).runOnce:261\n\tstorj.io/sto
rj/satellite/metainfo.(*Loop).Run:220\n\tstorj.io/storj/private/lifecycle.(*Group).Run.func2.1:81\n\truntime/pprof.Do:40\n\tstorj.io/storj/private/lifecycle.(*Group).Run.func2:80\n\tgolang.org/x/sync/errgroup.(*Group).Go.func1:57"}
                                                       | storj.io/storj/private/lifecycle.(*Group).Run.func2
                                                       |        D:/jskang/003-workspace/004-github/storj/storj/private/lifecycle/group.go:87
                                                       | golang.org/x/sync/errgroup.(*Group).Go.func1
                                                       |        C:/Users/kjuns/go/pkg/mod/golang.org/x/sync@v0.0.0-20201207232520-09787c993a3a/errgroup/errgroup.go:57
satellite-core/0                          14:59:53.048 | DEBUG  process/exec_conf.go:394        Unrecoverable error     {"error": "checker error: could not get estimate of total number of nodes: checker error: segment health is mean
ingless: there are no nodes", "errorVerbose": "checker error: could not get estimate of total number of nodes: checker error: segment health is meaningless: there are no nodes\n\tstorj.io/storj/satellite/repair/checker.(*checkerObse
rver).RemoteSegment:329\n\tstorj.io/storj/satellite/metainfo.(*observerContext).RemoteSegment:128\n\tstorj.io/storj/satellite/metainfo.handlePointer:324\n\tstorj.io/storj/satellite/metainfo.iterateDatabase.func2:406\n\tstorj.io/stor
j/storage/postgreskv.(*Client).IterateWithoutLookupLimit:233\n\tstorj.io/storj/satellite/metainfo.iterateDatabase:374\n\tstorj.io/storj/satellite/metainfo.(*Loop).runOnce:261\n\tstorj.io/storj/satellite/metainfo.(*Loop).Run:220\n\ts
torj.io/storj/private/lifecycle.(*Group).Run.func2.1:81\n\truntime/pprof.Do:40\n\tstorj.io/storj/private/lifecycle.(*Group).Run.func2:80\n\tgolang.org/x/sync/errgroup.(*Group).Go.func1:57"}
Error: exit status 1

```

**Comment by JunSungKang** [comment]
I solved it after initializing the schema tables in PostgreSQL.

I am now wondering what caused this problem, not how to fix it.

**Comment by egonelbre** [comment]
My best guess based on the information is that the postgres data was wiped somehow.

Did you start a docker container for postgres without specifying the mount directory? This would mean that it'll lose all content after restart.

**Comment by egonelbre** [comment]
Ah, now I remember... there's a known problem with storj-sim.

Currently when you take the satellite offline there won't be any storage node contact messages -- so when the satellite comes back online it thinks that all the storage nodes have been offline. So, the reliability checker properly shuts down, because it properly understands that "all nodes offline" is an invalid database state and it's unsafe to continue.

I think the fix was to adjust the "offline time" when some storage node is detected as being offline. See the config value `overlay.node.online-window` and set it do something higher.

**Comment by JunSungKang** [comment]
> Ah, now I remember... there's a known problem with storj-sim.
> 
> Currently when you take the satellite offline there won't be any storage node contact messages -- so when the satellite comes back online it thinks that all the storage nodes have been offline. So, the reliability checker properly shuts down, because it properly understands that "all nodes offline" is an invalid database state and it's unsafe to continue.
> 
> I think the fix was to adjust the "offline time" when some storage node is detected as being offline. See the config value `overlay.node.online-window` and set it do something higher.

Thanks for the info on the `overlay.node.online-window` option.

And i not use docker.

i in order to writer Master's Thesis, modifying source and test. : )
step 1. use `make install storj-sim`,
step 2. use `storj-sim network run`
step 3. use `gateway run`
step 4. use uplink binary

**Comment by egonelbre** [comment]
> i in order to writer Master's Thesis, modifying source and test

Note, the storj-sim and actual production server configurations do not match. storj-sim is intended for local testing and ensuring that you don't need the production servers. This means the number of storage nodes has been reduced (running 10k other processes usually isn't feasible). Similarly some of the services run more frequently and we have enabled some extra checks and verifications, to help us find bugs. There are also no load-balancers, multiple clusters, database clusters etc.

In other words verify that the configuration is suitable for what you intend to measure and test.

> step 3. use gateway run

As long as you have storj sim setup, it should start gateway for you already. We'll remove that at some point though.

> step 4. use uplink binary

gateway is not required for using uplink binary. gateway is only needed for making s3 like requests.
